src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1761,31 +1771,58 @@ private void VisitQueryableMethodCall(MethodCallExpression expression)
 
                     VisitExpression(expression.Arguments[0]);
                     break;
-                case "Where":
+                case "Filter":
+                {
+                    _insideFilter++;
+                    VisitExpression(expression.Arguments[0]);
+                    using (FilterModeScope(true))
                     {
-                        _insideWhere++;
-                        VisitExpression(expression.Arguments[0]);
                         if (_chainedWhere)
                         {
                             DocumentQuery.AndAlso();
                             DocumentQuery.OpenSubclause();
                         }
-                        if (_chainedWhere == false && _insideWhere > 1)
-                            DocumentQuery.OpenSubclause();
 
-                        if (expression.Arguments.Count == 3)
-                            _insideExact = (bool)GetValueFromExpression(expression.Arguments[2], typeof(bool));
+                        if (_chainedWhere == false && _insideFilter > 1)
+                            DocumentQuery.OpenSubclause();
 
                         VisitExpression(((UnaryExpression)expression.Arguments[1]).Operand);
-
-                        _insideExact = false;
-
-                        if (_chainedWhere == false && _insideWhere > 1)
+                        if (_chainedWhere == false && _insideFilter > 1)
                             DocumentQuery.CloseSubclause();
                         if (_chainedWhere)
                             DocumentQuery.CloseSubclause();
-                        _chainedWhere = true;
-                        _insideWhere--;
+                        _insideFilter--;
+                    }
+
+                    break;
+                }
+                case "Where":
```

Comment:
`nameof`?

Reply:
Everything we have there is just hard coded strings for method names. Let's not change that on 5.3 branch only to avoid unnecessary conflicts